src/main/java/com/github/sh0ckr6/achievementborder/listeners/WorldSetup.java

```diff
@@ -46,7 +54,9 @@ public WorldSetup(AchievementBorder plugin) {
   @EventHandler(priority = EventPriority.LOWEST)
   public void onPlayerJoin(PlayerJoinEvent event) {
     // Only continue if this is first time setup
-    if (!plugin.advancements.isEmpty()) return;
+    if (setupCompleted) return; // Check variable to see if setup has been completed.
+    setupCompleted = ConfigManager.readFromConfig("config", "setup-complete"); // Pull from config if we haven't completed setup.
+    if (setupCompleted) return; // Check the config's result and return if the config says we've completed setup.
```

Comment:
This can be one-lined
```suggestion
    if (ConfigManager.readFromConfig("config", "setup-complete")) return; // Check the config's result and return if the config says we've completed setup.
```